File: x-baseproject/src/main/java/com/xia/baseproject/rxbus/RxBusHelper.java
package com.xia.baseproject.rxbus;

import com.blankj.rxbus.RxBusManager;
import com.blankj.rxbus.RxBusMessage;

/**
 * @author xia
 * @date 2018/8/13.
 */
@SuppressWarnings("WeakerAccess")
public final class RxBusHelper {

    public static void post(final String tag) {
        post(new RxBusMessage(""), tag);
    }

    public static void post(final RxBusMessage event, final String tag) {
        RxBusManager.post(event, tag);
    }

    public static void postSticky(final String tag) {
        postSticky(new RxBusMessage(""), tag);
    }

    public static void postSticky(final RxBusMessage event, final String tag) {
        RxBusManager.postSticky(event, tag);
    }
}

File: fly/src/main/java/com/xia/fly/integration/rxbus/RxBusHelper.kt
package com.xia.fly.integration.rxbus

import com.xia.flyrxbus.RxBus
import com.xia.flyrxbus.RxBusManager
import com.xia.flyrxbus.RxBusMessage

/**
 * @author xia
 * @date 2018/8/13.
 */
@Suppress("unused")
object RxBusHelper {

    @JvmStatic
    fun post(tag: String) {
        post(RxBusMessage(""), tag)
    }

    @JvmStatic
    fun post(event: RxBusMessage, tag: String) {
        RxBusManager.post(event, tag)
    }

    @JvmStatic
    fun postSticky(tag: String) {
        postSticky(RxBusMessage(""), tag)
    }

    @JvmStatic
    fun postSticky(event: RxBusMessage, tag: String) {
        RxBusManager.postSticky(event, tag)
    }

    @JvmStatic
    fun removeSticky(event: Any) {
        RxBusManager.removeSticky(event)
    }

    @JvmStatic
    fun removeSticky(event: Any, tag: String) {
        RxBusManager.removeSticky(event, tag)
    }

    @JvmStatic
    fun subscribeWithTags(subscriber: Any,
                          callback: IRxBusCallback?,
                          vararg tags: String) {
        RxBusManager.subscribeWithTags(subscriber, object : RxBus.Callback<RxBusMessage>() {
            override fun onEvent(tag: String, t: RxBusMessage) {
                callback?.callback(tag, t)
            }
        }, *tags)
    }

    @JvmStatic
    fun subscribeStickyWithTags(subscriber: Any,
                                callback: IRxBusCallback?,
                                vararg tags: String) {
        RxBusManager.subscribeStickyWithTags(subscriber, object : RxBus.Callback<RxBusMessage>() {
            override fun onEvent(tag: String, t: RxBusMessage) {
                callback?.callback(tag, t)
            }
        }, *tags)
    }

    @JvmStatic
    fun unregister(subscriber: Any?) {
        RxBusManager.unregister(subscriber)
    }
}
